### PR TITLE
EVAKA-4203: authors section and section numberings

### PR DESCRIPTION
**Pull request by Joosakur**

### CODE REVIEW

File: frontend/src/employee-frontend/components/vasu/VasuPage.tsx [hunk 1]
```diff
@@ -182,7 +190,10 @@ export default React.memo(function VasuPage({
   ) {
     return (
       <>
-        <Label>{question.name}</Label>
+        <Label>
+          {sectionIndex + 1 + dynamicSectionsOffset}.{questionIndex + 1}{' '}
```
Comment (kimifoud) [inline frontend/src/employee-frontend/components/vasu/VasuPage.tsx:194]: Consider extracting this repeated logic

```ts
  const getDynamicQuestionNumber = (
    sectionIndex: number,
    questionIndex: number
  ) => `${sectionIndex + 1 + dynamicSectionsOffset}.${questionIndex + 1}`
```

File: frontend/src/employee-frontend/components/vasu/VasuPage.tsx [hunk 2]
```diff
@@ -317,11 +336,19 @@ export default React.memo(function VasuPage({
...
+            setContent={setAuthorsContent}
+          />
+          <Gap size={'L'} />
           {content.sections.map((section, sectionIndex) => {
             return (
               <Fragment key={section.name}>
                 <ContentArea opaque>
-                  <h2>{section.name}</h2>
+                  <H2>
+                    {sectionIndex + 2}. {section.name}
```
Comment (kimifoud) [inline frontend/src/employee-frontend/components/vasu/VasuPage.tsx:350]: Maybe use `dynamicSectionsOffset` here too.

File: frontend/src/employee-frontend/components/vasu/sections/AuthorsSection.tsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+        // add empty row at end
+        if (!authorIsEmpty(otherAuthors[otherAuthors.length - 1])) {
+          otherAuthors.push({ name: '', title: '', phone: '' })
+        }
+
+        return { ...prev, otherAuthors }
+      })
+    }
+  }, [content, setContent])
```
Comment (kimifoud) [inline frontend/src/employee-frontend/components/vasu/sections/AuthorsSection.tsx:48]:
```suggestion
  }, [content.otherAuthors, setContent])
```

File: frontend/src/employee-frontend/components/vasu/sections/AuthorsSection.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                }
+              />
+            </FixedSpaceColumn>
+
+            <FixedSpaceColumn spacing="xxs">
+              <Label>{t.authorInfo.phone}</Label>
+              <InputField
+                value={author.phone}
+                onChange={(value) =>
+                  setContent((prev) => ({
```
Comment (kimifoud) [inline frontend/src/employee-frontend/components/vasu/sections/AuthorsSection.tsx:165]: This repeated onChange handler could be DRY'ed.
```ts
const onChange = (key: keyof AuthorInfo) => (value: string) =>
  setContent((prev) => ({
    ...prev,
    otherAuthors: [
      ...prev.otherAuthors.slice(0, i),
      {
        ...prev.otherAuthors[i],
        [key]: value
      },
      ...prev.otherAuthors.slice(i + 1)
    ]
  }))
```

```tsx
 <InputField value={author.name} onChange={onChange('phone')} />
 ```

Comment (kimifoud) [reply frontend/src/employee-frontend/components/vasu/sections/AuthorsSection.tsx:165]: Same goes for primary author.

**Review by kimifoud — CHANGES_REQUESTED** [review]

File: frontend/src/employee-frontend/components/common/VasuStateChip.tsx [hunk 5]
```diff
@@ -22,5 +22,9 @@ interface StateChipProps {
 }
 
 export function VasuStateChip({ labels, state }: StateChipProps) {
-  return <StaticChip color={vasuStateChip[state]}>{labels[state]}</StaticChip>
+  return (
+    <StaticChip color={vasuStateChip[state]} textColor={colors.greyscale.white}>
```
[the remark this answers is not in the document]
Comment (Joosakur) [reply frontend/src/employee-frontend/components/common/VasuStateChip.tsx:26]: it depends on the background color so can't

[the remark this answers is not in the document]
Comment (Joosakur) [reply frontend/src/employee-frontend/components/common/VasuStateChip.tsx:26]: There is some library which tries to guess the better text color from contrast, but it doesn't seem to work too well :D

```
  color: ${({ theme: { colors }, ...p }) =>
    p.textColor ??
    readableColor(p.color, colors.greyscale.darkest, colors.greyscale.white)};
```

[the remark this answers is not in the document]
Comment (kimifoud) [reply frontend/src/employee-frontend/components/common/VasuStateChip.tsx:26]: Well, if we want to fix the problem in all places, the least we could do is create a color override mapping in `StaticChip` for the problematic accent colors.

**Review by kimifoud — APPROVED** [review]
👍

File: frontend/src/employee-frontend/components/common/VasuStateChip.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Joosakur) [reply frontend/src/employee-frontend/components/common/VasuStateChip.tsx:26]: yeah that might be a good idea 🤔 won't botter in this PR though 😄

**Pull request merged**